### PR TITLE
Improved README.

### PR DESCRIPTION
**Pull request by StarWitch**
* Expands upon what both Taisei and Touhou are
* Removes technical stuff that will be added in future PRs (game controller, etc)
* Removes freenode link (RIP)

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -6,126 +6,122 @@ Taisei
...
-- ``OES_vertex_array_object``
-- ``EXT_frag_depth``
-- ``EXT_instanced_arrays`` or ``ANGLE_instanced_arrays`` or
-  ``NV_instanced_arrays``
+-  OpenSSL (for a better SHA-256 implementation; used in shader cache)
+-  SPIRV-Cross >= 2019-03-22 (for OpenGL ES backends)
+-  libshaderc (for OpenGL ES backends)
+-  ANGLE (for ANGLE support, required for Windows)
+-  GameMode headers (Linux only; for automatic `GameMode
+   <https://github.com/FeralInteractive/gamemode>`__ integration)
```
Comment (Akaricchi) [inline README.rst:120]: Could of things here

* I don't like the merging of runtime dependencies with build-time dependencies. The distinction is pretty important particularly in a cross-compilation environment.
* cglm is now missing (runtime dependency)
* python-zstandard is also missing (build dependency)
* ANGLE is not "required" on windows. For the purposes of hacking on the codebase, you probably won't even need it, unless you're stuck with an Intel iGPU.
* I think we can just remove mentions of OpenGL ES 2.0, and simply state that GL3.3+ or ES3.0+ is required. The gles20 backend is unmaintained and doesn't really work; at best it'll only work on ES3-capable hardware.

File: README.rst [hunk 2]
```diff
@@ -6,126 +6,122 @@ Taisei
...
 
-The OpenGL ES 3.0 backend is not built by default. To enable it, do:
+If you want to pull them in manually, be sure to run the following:
 
 ::
 
-    meson configure -Dr_gles30=true -Dshader_transpiler=true
+    cd taisei
+    git submodule init
+    git submodule update
```
Comment (Akaricchi) [inline README.rst:89]: Actually you can just do `git submodule update --init` with non-ancient git.

File: README.rst [hunk 3]
```diff
@@ -6,126 +6,122 @@ Taisei
...
+
+You should also run ``git submodule update`` whenever you pull in
+new code, checkout another branch, etc. The ``pull`` and ``checkout`` helper
+scripts can do that for you automatically.
 
-See `here <doc/ENVIRON.rst>`__ for information on how to activate it.
-Alternatively, do this to make GLES 3.0 the default backend:
+For more in-depth development and compiling instructions, see our
+`Development Guide <doc/DEVELOPMENT.rst>`__.
```
Comment (Akaricchi) [inline README.rst:96]: I think it's very important to have a super basic TL;DR version of how to build and run the thing, i.e. literally
```sh
meson setup build
meson compile -C build
meson install -C build
```
Asking a potential new user/contributor to read a book that doesn't even exist yet is not the best look. On that note, I'd also remove references to not-yet-written documents, and have them added back in when they are ready.

Would be even better if all the info on how to properly obtain the source code could be further condensed somehow. Perhaps it's not necessary to mention the alternative way to init submodules… teaching people how git works is not a goal here.

File: README.rst [hunk 4]
```diff
@@ -6,126 +6,122 @@ Taisei
...
 
-Build-only dependencies
-^^^^^^^^^^^^^^^^^^^^^^^
-
--  Python >= 3.6
--  meson >= 0.53.0
--  [python-zstandard](https://github.com/indygreg/python-zstandard)
-
-Optional:
+Taisei Project is highly portable, and is written in C (specifically, C11),
```
Comment (Akaricchi) [inline README.rst:18]: Probably can just say written in C11.

**Comment by StarWitch** [comment]
Fixed all the PR comments, and added the icon to the header too. Let me know how it looks.

The further dev instructions/troubleshooting links will get added back in future PRs.

File: README.rst [hunk 5]
```diff
@@ -1,131 +1,156 @@
...
 -  zlib
 
-Optional:
+Optional
+********
 
 -  OpenSSL (for a better SHA-256 implementation; used in shader cache)
 -  SPIRV-Cross >= 2019-03-22 (for OpenGL ES backends)
 -  libshaderc (for OpenGL ES backends)
+-  `ANGLE <https://github.com/google/angle>`__ (for ANGLE support, useful for Windows builds)
```
Comment (Akaricchi) [inline README.rst:94]: `ANGLE (for ANGLE support)` is pretty tautological. Maybe better to say that it's usable via the gles30 backend and/or that it's for rendering on non-OpenGL platforms?

File: README.rst [hunk 6]
```diff
@@ -1,131 +1,156 @@
...
-Build-only dependencies
-^^^^^^^^^^^^^^^^^^^^^^^
+Build-Time Dependenices
+_______________________
 
 -  Python >= 3.6
--  meson >= 0.53.0
--  [python-zstandard](https://github.com/indygreg/python-zstandard)
+-  `python-zstandard <https://github.com/indygreg/python-zstandard>`__ >= 0.11.1
+-  meson >= 0.56.0
```
Comment (Akaricchi) [inline README.rst:103]: We still support >=0.53.0 for regular builds, for now. It's just that our CI machine files require some >=0.56.0 features

File: README.rst [hunk 7]
```diff
@@ -1,131 +1,156 @@
...
+Obtaining Source Code
+^^^^^^^^^^^^^^^^^^^^^
 
-Latest code from git
-""""""""""""""""""""
+You can obtain the source code of the stable releases from the
+`Releases <https://github.com/taisei-project/taisei/releases>`__ page on
+GitHub, alongside the binaries. (Note that you **MUST** use the releases page,
+as the ``Download ZIP`` link on the main repo *does not* include all the code
+necessary to build the game.)
```
Comment (Akaricchi) [inline README.rst:113]: I think it's good to say that it doesn't include submodules, specifically, to provide some context as to why this is the case to the more git savvy reader. I would say something like "as Github does not automatically include the necessary submodules in those packages", to imply that this mess isn't due to our choice.

File: README.rst [hunk 8]
```diff
@@ -1,131 +1,156 @@
...
-If you cloned Taisei from git, make sure the submodules are initialized:
+You can also grab the latest code from git using the following commands:
 
 ::
 
-    git submodule init
-    git submodule update
+    git clone https://github.com/taisei-project/taisei
+    cd taisei
+    git submodule update --init --recursive
```
Comment (Akaricchi) [inline README.rst:121]: When I said:
> Would be even better if all the info on how to properly obtain the source code could be further condensed somehow. Perhaps it's not necessary to mention the alternative way to init submodules… teaching people how git works is not a goal here.

I actually meant to just keep the `git clone` form:
```suggestion
    git clone --recurse-submodules https://github.com/taisei-project/taisei
```
i.e. `git submodule update --init` is what I referred to as the "alternative way".

File: README.rst [hunk 9]
```diff
@@ -1,131 +1,156 @@
...
+You should also run ``git submodule update`` whenever you pull in
+new code, checkout another branch, or perform any ``git`` actions. The ``pull``
+and ``checkout`` helper scripts can do that for you automatically.
 
-**Important:** You should also run ``git submodule update`` whenever you pull in
-new code, checkout another branch, etc. The ``pull`` and ``checkout`` helper
-scripts can do that for you automatically.
+**Important:** Again, make sure you download the source code from either the
+``Releases`` page *or* using ``git clone``. The ``Download ZIP`` link *will not
+work!*
```
Comment (Akaricchi) [inline README.rst:129]: I don't think it's necessary to repeat this twice. Let's trust the reader's attention a little.

File: README.rst [hunk 10]
```diff
@@ -1,131 +1,156 @@
...
 
-Compiling from source
+Compiling Source Code
 ^^^^^^^^^^^^^^^^^^^^^
 
-To build and install Taisei on \*nix, just follow these steps:
+Currently, we recommend building Taisei on a \*nix or macOS-based system.
```
Comment (Akaricchi) [inline README.rst:134]: "POSIX-like" would cover them all (and even some obscure ones like Haiku — and yes, Taisei works there without modifications; no hardware acceleration though :smile:)

File: README.rst [hunk 11]
```diff
@@ -1,131 +1,156 @@
...
-    $prefix/taisei
-    $prefix/data/
+    meson setup build/
+    meson compile -C build/
+    meson install -C build/
 
-``install_relative`` is always set when building for Windows.
-
-The OpenGL ES 3.0 backend is not built by default. To enable it, do:
+You can also package the game into a ``.zip`` archive.
```
Comment (Akaricchi) [inline README.rst:145]: I don't think it's necessary to document this here; this is a detail of the release workflow that most users won't care about. If we ever want to document releases, that can go there (along with the other packing targets e.g. tar/txz).

**Review by Akaricchi — DISMISSED** [review]

File: README.rst (hunk 9 above)
Comment (StarWitch) [reply README.rst]: A bold strategy...

**Review by Akaricchi — APPROVED** [review]

**Pull request merged**